flow.py

```diff
@@ -3,47 +3,59 @@ class Flow(object):
     hosts and routers.
 
     Args:
-        ns: Instance of the NetworkSimulator class
-        flow_id: Unique id identifying the flow
+        ns (NetworkSimulator): Instance of the NetworkSimulator class
+        flow_id (int): Unique id identifying the flow
 
     Attributes:
-        flow_id: Unique id identifying the flow
-        src: The flow's source node
-        dest: The flow's destination node
-        data_amount: Data capacity of the flow
-
-        window_size: The size of the window
-        unacknowledged_packets: The list of packets with no acknowledgement
-        current_packet: The current packet in the flow
-        num_packets:  The number of packets that have been through the flow
-        last_acknowledged: The last acknowledged packet in the flow
+        flow_id (int): Unique id identifying the flow
+        src (Node): The flow's source node
+        dest (Node): The flow's destination node
+        data_amount (int): Data capacity of the flow
+
+        window_size (int): The size of the window
+        unacknowledged_packets (dict): The list of packets with no acknowledgement
+        current_packet (Packet): The current packet in the flow
+        num_packets (int):  The number of packets that has been sent so far
+        last_acknowledged (Packet): The last acknowledged packet in the flow
+
     """
-    def __init__(self, ns, flow_id):
+
+    def __init__(self, ns, flow_id, src, dest, data_amount=0, window_size=0, unacknowledged_packets=[],
+    current_packet=0, num_packets=0, last_acknowledged=0):
+        self.ns = ns
         self._flow_id = flow_id
+        self._src = src
+        self._dest = dest
+        self.data_amount = data_amount
+        self.window_size
+        self.unacknowledged_packets = unacknowledged_packets
+        self.current_packet = current_packet
+        self.num_packets = num_packets
+        self.last_acknowledged
 
     @property
     def flow_id(self):
         return self.flow_id
 
     @flow_id.setter
     def flow_id(self, flow_id):
-        raise Exception("Error: Can't change id of a flow.")
+        raise AttributeError("Error: Can't change id of a flow.")
```

Comment:
Since you're already raising an error you don't need to say error again. Just make it "Cannot modify flow id" or something.